cypress/support/commands.ts

```diff
@@ -24,17 +21,18 @@ Cypress.Commands.add('loadPage', (pageName) => {
   });
 });
 
-Cypress.Commands.add('iframe', (selector: string, options?: IframeOptions) => {
-  const DEFAULT_OPTIONS = {
-    // Cypress default timeout
-    timeout: 15000,
-  };
-
-  const ops = { ...DEFAULT_OPTIONS, ...options };
+Cypress.Commands.add('getIframeBody', (selector: string) => {
+  return cy
+    .get(selector, { timeout: 15000 })
+    .its(`0.contentDocument.body`)
+    .should('not.be.empty')
+    .then(cy.wrap);
+});
 
+Cypress.Commands.add('getIframeHtml', (selector: string) => {
   return cy
-    .get(selector, { timeout: ops.timeout })
-    .its(`0.contentDocument`)
+    .get(selector, { timeout: 15000 })
+    .its('0.contentDocument.documentElement')
```

Comment:
The reason some of these tests were flaky is bc we were looking for the `html` element within the iframe, but that element isn't natively available within an iframe, we can use this `documentElement`:

> Document.documentElement returns the Element that is the root element of the document (for example, the <html> element for HTML documents).

https://developer.mozilla.org/en-US/docs/Web/API/Document/documentElement